src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -454,19 +459,28 @@ private static boolean skipDoubleBraceInstInit(Details details) {
      * @return true if block has single-line format and is alone on a line.
      */
     private static boolean isBlockAloneOnSingleLine(Details details) {
-        final DetailAST rcurly = details.rcurly;
-        final DetailAST lcurly = details.lcurly;
         DetailAST nextToken = details.nextToken;
-        while (nextToken.getType() == TokenTypes.LITERAL_ELSE) {
+        while (nextToken != null && nextToken.getType() == TokenTypes.LITERAL_ELSE) {
             nextToken = Details.getNextToken(nextToken);
         }
-        if (nextToken.getType() == TokenTypes.DO_WHILE) {
+
+        if (nextToken != null && nextToken.getType() == TokenTypes.DO_WHILE) {
             final DetailAST doWhileSemi = nextToken.getParent().getLastChild();
             nextToken = Details.getNextToken(doWhileSemi);
         }
-        return TokenUtil.areOnSameLine(rcurly, lcurly)
+        boolean isBlockAloneOnSingleLine = false;
+        if (nextToken != null) {
+            final DetailAST rcurly = details.rcurly;
+            final DetailAST lcurly = details.lcurly;
+            isBlockAloneOnSingleLine = TokenUtil.areOnSameLine(rcurly, lcurly)
                 && (!TokenUtil.areOnSameLine(rcurly, nextToken)
                 || isRightcurlyFollowedBySemicolon(details));
+        }
+        if (nextToken == null
+                && TokenUtil.areOnSameLine(details.lcurly, details.rcurly)) {
+            isBlockAloneOnSingleLine = true;
+        }
```

Comment:
this 2 ifs can be simplified to something like
```
return TokenUtil.areOnSameLine(details.lcurly, details.rcurly) && (nextToken == null || !TokenUtil.areOnSameLine(rcurly, nextToken)
                || isRightcurlyFollowedBySemicolon(details))
```

Reply:
Done.